sites/game-client/ui/overlay.mjs

```diff
@@ -141,7 +145,10 @@ class Overlay {
    * This method can be implemented in the derived class if the derived class
    * wishes to handle this event.
    */
-  onPostHide() { return true; }
+  onPostHide() {
+    this.isOpen = false;
```

Comment:
There is no guarantee that the derived class will call the base class' onPostHide(). To better maintain this.isOpen, it would be recommended to put the update code in show() and hide() above.